src/evm-block-extractor/src/main.rs

```diff
@@ -157,7 +215,46 @@ fn write_blocks(
     Ok(())
 }
 
+/// Scans the output dir for existing block files and returns the last block number
+fn get_last_block_number_from_output_dir(output_dir: &Path) -> Option<u64> {
+    // scan files in output dir
+    let mut last_block: Option<u64> = None;
+    let mut output_dir = std::fs::read_dir(output_dir).ok()?;
+    for file in output_dir.by_ref() {
+        let file = file.ok()?;
+        log::info!("found file: {}", file.path().display());
+        last_block = match (
```

Comment:
this is overcomplicated. use `let mut last_block = 0;` and on every loop simply do `last_block = min(last_block, new_last_block)`